src/main.cpp

```diff
@@ -267,9 +286,12 @@ int main(int argc, char** argv)
     } else {
         // open perf.data in current CWD, if it exists
         // this brings hotspot closer to the behavior of "perf report"
-        const auto perfDataFile = QStringLiteral("perf.data");
-        if (QFile::exists(perfDataFile) && window) {
-            window->openFile(perfDataFile);
+        const auto perfDataFiles = findPerfDataFiles();
+        for (const auto& perfDataFile : perfDataFiles) {
```

Comment:
`perf report` opens only a single file and I have _lots_ of perf files in my directories...
I suggest to add either an optional integer limit or `boolean newestFileOnly` attribute to that function and use it here and above.

Reply:
the loop breaks after the first opened file, so this is fine as-is

Reply:
In theory this means that the caller may collect lots of file entries, just to drop all of those later.
That's not of a big issue currently as `findPerfDataFiles()` does not match our file open window (which I think it should, I'm creating a new issue to track it).